### PR TITLE
Add a test case for findDanglingImpls

### PR DESCRIPTION
**Pull request by alanwaketan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61104**

Summary:
This patch added a new test case for findDanglingImpls. The test case introduces a C++ extension which has a dangling impl such that findDanglingImpls can find it and output its information.

Test Plan:
python test/test_dispatch.py TestDispatch.test_find_dangling_impls_ext

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29512520](https://our.internmc.facebook.com/intern/diff/D29512520)

### CODE REVIEW

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29512520).

**Comment by alanwaketan** [comment]
This is a follow up for d5be67a338f1bb8762c49e4be139689631f71895.

**Comment by alanwaketan** [comment]
Oops, didn't know Windows uses '\' instead of '/'. (I guess I never need to deal with Windows in my previous job. 🤪) Will fix that.

But I'm a little bit surprised by the Linux/CUDA failure. Not sure why it has two dangling implementations.

File: test/test_dispatch.py [hunk 1]
```diff
@@ -755,6 +757,32 @@ def test_overwrite_math(self):
...
+        self.assertEqual(1, len(impls))
+        self.assertEqual(
+            '''\
+name: __test::foo
+schema: (none)
+CPU: registered at {}:5 :: () -> () [ boxed unboxed ]
+'''.format(extension_path),
+            impls[0])
+
+
```
Comment (alanwaketan) [inline test/test_dispatch.py:784]: These empty lines were added by accidents.

**Review by ezyang — APPROVED** [review]
Very nice, thanks!

**Comment by alanwaketan** [comment]
> Very nice, thanks!

Thanks for approving this patch, Ed! However, wondering if you have any insights about the following CI failures:
1. pytorch_linux_xenial_cuda11_1_cudnn8_py3_gcc7_test1.
2. pytorch_linux_bionic_cuda10_2_cudnn7_py3_9_gcc7_test2.

Not sure why there is one more dangling impl. @ezyang

**Comment by ezyang** [comment]
well if you had printed the output of findDanglingImpls that would have given you a good clue right :)

**Comment by alanwaketan** [comment]
> well if you had printed the output of findDanglingImpls that would have given you a good clue right :)

Thanks, Ed. That's right. It looks like the dangling implementation is unrelated to my patch. I just took a look at the full log. It indicates that there is one dangling implementation for the default test. Let me modify the default test case such that it always prints the info.

**Comment by alanwaketan** [comment]
@alanwaketan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29512520).

**Pull request closed without merging**